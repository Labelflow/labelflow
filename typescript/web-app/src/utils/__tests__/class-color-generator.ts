import {
  previousHexToNextHexMap,
  getNextClassColor,
} from "../class-color-generator";

test("Class color generator mapping has the right value", () => {
  expect(previousHexToNextHexMap).toMatchSnapshot();
});

it("Gets the next hex color from previous valid hex color", () => {
  expect(getNextClassColor("#65A30D")).toEqual("#0D9488");
});

it("It throws when a non valid hex color is passed", () => {
  expect(() => getNextClassColor("#000000")).toThrow();
});
