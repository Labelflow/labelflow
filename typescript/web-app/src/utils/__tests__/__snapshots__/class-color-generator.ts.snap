// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Class color generator mapping has the right value 1`] = `
Object {
  "#0284C7": "#7C3AED",
  "#059669": "#0284C7",
  "#0891B2": "#4F46E5",
  "#0D9488": "#2563EB",
  "#16A34A": "#0891B2",
  "#22D3EE": "#818CF8",
  "#2563EB": "#9333EA",
  "#2DD4BF": "#60A5FA",
  "#34D399": "#38BDF8",
  "#38BDF8": "#A78BFA",
  "#4ADE80": "#22D3EE",
  "#4F46E5": "#C026D3",
  "#60A5FA": "#C084FC",
  "#65A30D": "#0D9488",
  "#7C3AED": "#F472B6",
  "#818CF8": "#E879F9",
  "#9333EA": "#FB7185",
  "#A3E635": "#2DD4BF",
  "#A78BFA": "#DB2777",
  "#C026D3": "#F87171",
  "#C084FC": "#E11D48",
  "#CA8A04": "#059669",
  "#D97706": "#16A34A",
  "#DB2777": "#FB923C",
  "#DC2626": "#CA8A04",
  "#E11D48": "#D97706",
  "#E879F9": "#DC2626",
  "#EA580C": "#A3E635",
  "#F472B6": "#EA580C",
  "#F87171": "#FACC15",
  "#FACC15": "#34D399",
  "#FB7185": "#FBBF24",
  "#FB923C": "#65A30D",
  "#FBBF24": "#4ADE80",
}
`;
